### PR TITLE
 Add Person, Student and Teacher classes

### PR DESCRIPTION
**Pull request by Metsanpeitto**
### Project requirements
- Create class `Person` with  the following:
  - Instance vars: `@id`, `@name`, and `@age`.
  - Constructor with `name`,  `age`, and `parent_permission` as parameter. `name` and `parent_permission` are optional and have default values of `"Unknown"` and `true`.
  - Getters for `@id`, `@name`, and `@age`.
  - Setters for `@name` and `@age`.
  - Private method `is_of_age?` that returns `true` if `@age` is greater or equal to 18 and `false` otherwise.
  - Public method `can_use_services?` that returns `true` if person is of age or if they have permission from parents.
- Create class `Student` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructar by adding `@classroom` and a parameter for it.
  - Method `play_hooky` that returns `"¯\(ツ)/¯"`.
- Create class `Teacher` with the following:
  - Inherits from `Person`.
  - Constructor extends parent's constructar by adding `@specialization` and a parameter for it.
  - Override `can_use_services?` so it always returns `true`.
- Each class should be saved in a separate file.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    @name = name
+  end
+
+  def age_set(age)
+    @name = age
+  end
+
+  def can_use_services?
+    is_of_age? && parent_permission == true
+  end
```
Comment (arslanbisharat) [inline person.rb]: - [ ] The requirements for this project requested to have either age or parent permission true. Let's make the changes. 
Capture

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    @age
+  end
+
+  def name_set(name)
+    @name = name
+  end
+
+  def age_set(age)
+    @name = age
+  end
```
Comment (arslanbisharat) [inline person.rb]: - [ ] name and age instance variables are the only ones that should be able to be modified that's why they are defined as attr_accessors (here both Getters and Setters are defined)

`  attr_accessor :name, :age `

- [ ] id should only be able to be retrieved so only the Getter is defined

`  attr_reader :id 
`
Capture

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class Teacher < Person
+  def initialize(specialization)
+    super
+    @specialization = specialization
+  end
```
Comment (arslanbisharat) [inline teacher.rb]: - [ ] The requirements requested you to extend the parent's constructor by adding `@specialization `. Let's add other parameters here too. 
Capture

File: teacher.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+  def play_hookey
+    "¯\(ツ)/¯"
+  end
```
Comment (arslanbisharat) [inline teacher.rb]: - [ ] Can you please explain why this block of code is present here? I cannot see any requirement related to this in the project.

File: student.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+class Student < Person
+  def initialize(classroom)
+    super
+    @classroom = classroom
+  end
```
Comment (arslanbisharat) [inline student.rb]: - [ ] The requirements requested you to extend the parent's constructor by adding `@classroom  `. Let's add other parameters here too. 
Capture

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: person.rb (hunk 1 above)
Comment (Metsanpeitto) [reply person.rb]: :heavy_check_mark: Fixed

File: features/create_rental.rb [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+  list_books
+  book_index = gets.chomp
+
+  Texts.text_cr2
+  list_people
+  people_index = gets.chomp
+  book = @books[book_index.to_i]
+  person = @people[people_index.to_i]
+  create_rental(book.title, person[:name])
+end
```
Comment (arslanbisharat) [inline features/create_rental.rb]: - [ ] Why have you added this file? I cannot see any requirements related to this.

File: README.md [hunk 7]
```diff
@@ -1 +1,112 @@
 # ruby_oop-school-library_m4_w3
+
+![Microverse](https://img.shields.io/badge/Microverse-blueviolet)
+
+"Ruby OOP school library" is  a tool to record what books are in the library and who borrows them. The app  allow you to:
```
Comment (arslanbisharat) [inline README.md:5]: - [ ] Let's add  .vscode to the gitignore file so we have a clean repository.

**Review by arslanbisharat — CHANGES_REQUESTED** [review]

File: person.rb (hunk 2 above)
Comment (Metsanpeitto) [reply person.rb]: :heavy_check_mark: Fixed!

File: teacher.rb (hunk 3 above)
Comment (Metsanpeitto) [reply teacher.rb]: :+1:

File: teacher.rb (hunk 4 above)
Comment (Metsanpeitto) [reply teacher.rb]: :heavy_check_mark: Fixed!

File: student.rb (hunk 5 above)
Comment (Metsanpeitto) [reply student.rb]: :heavy_check_mark: Fixed!

File: features/create_rental.rb (hunk 6 above)
Comment (Metsanpeitto) [reply features/create_rental.rb]: :heavy_check_mark: Ok, I misplaced it. Fixed now

File: README.md (hunk 7 above)
Comment (Metsanpeitto) [reply README.md:5]: :+1:

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
# Changes Requested, 

Hi there, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

 ## Required Changes ♻️
- [ ] As requested in the previous review performed by me, please add .vscode to gitignore so we have a clean repository without unnecessary files. 
Capture

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have limited number of code reviews per project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Comment by Metsanpeitto** [comment]
Ok. I fixed it.

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

To Highlight: 
- The Pull Request has a proper title and description. ✔️ 
- You have added a descriptive Readme file. Good Job. ✔️ 
- There is no linter error is present. ✔️ 
- Project has met all the technical aspects you needed for this project. Keep rocking! ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**